dev/io.openliberty.org.apache.jasper.expressionLanguage.5.0_fat/test-applications/EL50BasicTests.war/src/io/openliberty/el50/fat/basic/servlets/EL50BasicTestsServlet.java

```diff
@@ -0,0 +1,166 @@
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.el50.fat.basic.servlets;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import jakarta.el.ELContext;
+import jakarta.el.ELManager;
+import jakarta.el.ELProcessor;
+import jakarta.el.ELResolver;
+import jakarta.el.ExpressionFactory;
+import jakarta.el.StandardELContext;
+import jakarta.el.ValueExpression;
+import jakarta.servlet.annotation.WebServlet;
+
+
+/**
+ * Servlet for testing basic changes in EL 5.0
+ */
+@WebServlet({ "/EL50BasicTests" })
+public class EL50BasicTestsServlet extends FATServlet {
+    private static final long serialVersionUID = 1L;
+
+    ELProcessor elp;
+
+    public EL50BasicTestsServlet() {
+        super();
+
+        elp = new ELProcessor();
+        elp.defineBean("simpleBean", new SimpleBean());
+    }
+
+    /**
+     * 
+     * Expression Language 5.0 in Jakarta EE10 added generics to methods in multiple classes https://github.com/jakartaee/expression-language/issues/157
+     * Classes with changes we tests are ELProcessor, ExpressionFactory, ELContext, ValueExpression, ELResolver (returns null)
+     * Classes with changes we don't test or implicitly test TypeConverter(returns null), CompositeELResolver(returns null)
+     *
+     * @throws Exception
+     */
+    @Test
+    @Mode(TestMode.FULL)
+    public void testGenerics() throws Exception {
+        // Test generics with ELProcessor uses ValueExpression.getValue under the covers that uses generics
+        // No need to cast this to boolean like previous versions
+        assertTrue("The type was expected to be true for ELProcessor but was not", elp.eval("simpleBean.getNumber() == 25"));
+
+        // Test generics with ExpressionFactory
+        ExpressionFactory factory = ExpressionFactory.newInstance();
+        Integer number = 12;
+        Double result = factory.coerceToType(number, java.lang.Double.class);
+        assertNotNull(result);
+        assertTrue("The type was expected to be coerced to Double for ExpressionFactory but was not: " + result, result.equals(new Double(number)));
+
+        // Test generics with ELContext uses ELResolver.convertToType under the covers
+        ELContext elctx = new StandardELContext(ELManager.getExpressionFactory());
+        elctx.convertToType(number, java.lang.Double.class);
+        assertNotNull(result);
+        assertTrue("The type was expected to be converted to Double for ELContext but was not: " + result, result.equals(new Double(number)));
+
+        // Test generics with ValueExpression technically not necessary since it is used in the covers of elprocessor
```

Comment:
"used within the internals of ELProcessor" does that sound like a better description?

Reply:
Does sounds much better. I intended it to be `under the covers` not sure how it ended `in the covers` but I'll make the change